django/db/models/options.py

```diff
@@ -566,6 +571,26 @@ def get_field(self, field_name):
         except KeyError:
             raise FieldDoesNotExist("%s has no field named '%s'" % (self.object_name, field_name))
 
+    def get_field_by_path(self, path):
```

Comment:
If we're going to add this, is there an opportunity to share the logic, rather than just copy from `get_fields_by_path()`? ("I've seen this before...", I thought.)

Reply:
it is from [get_fields_from_path](https://github.com/django/django/blob/da92ec79621fc0bba671d8afa52b7f6884962fe5/django/contrib/admin/utils.py#L482) in `contrib/admin/utils.py`